lightning/src/ln/channelmanager.rs

```diff
@@ -9050,7 +9112,37 @@ where
 		let hmac = payment_id.hmac_for_offer_payment(nonce, expanded_key);
 		let context = OffersContext::OutboundPayment { payment_id, nonce, hmac: Some(hmac) };
 		let reply_paths = self.create_blinded_paths(context)
-			.map_err(|_| Bolt12SemanticError::MissingPaths)?;
+			.map_err(|_| Bolt12CreationError::BlindedPathCreationFailed)?;
+
+		let total_liquidity: u64 = {
+			let per_peer_state = &self.per_peer_state.read().unwrap();
+			let mut sum: u64 = 0;
+
+			per_peer_state.iter().for_each(|(_, peer_state_mutex)| {
+				let peer_state_lock = peer_state_mutex.lock().unwrap();
+				let peer_state = &*peer_state_lock;
+
+				peer_state.channel_by_id.iter().for_each(|(_, phase)| {
+					sum += phase.context().get_available_balances(&self.fee_estimator).outbound_capacity_msat;
+				});
```

Comment:
nit: Instead of `for_each()`, you could use `map().sum()` and add the result to `sum`.